awscli/customizations/s3/subcommands.py

```diff
@@ -1008,6 +1024,37 @@ def run(self):
             rc = 2
         return rc
 
+    def _get_s3_transfer_handler(self, result_queue):
```

Comment:
Cool. I was thinking that an abstraction was eventually going to be needed. Just was not sure when, but I'm totally with creating the abstraction now.